airflow/providers/fab/auth_manager/api/auth/backend/basic_auth.py

```diff
@@ -21,14 +21,17 @@
 from functools import wraps
 from typing import TYPE_CHECKING, Any, Callable, TypeVar, cast
 
-from flask import Response, request
+from flask import Response, has_app_context, request
 from flask_appbuilder.const import AUTH_LDAP
 from flask_login import login_user
 
 from airflow.providers.fab.auth_manager.security_manager.override import FabAirflowSecurityManagerOverride
 from airflow.www.extensions.init_auth_manager import get_auth_manager
 
 if TYPE_CHECKING:
+    from fastapi.security import HTTPBasicCredentials
+    from werkzeug.datastructures import Authorization
```

Comment:
Yes this adds changes to the provider and core. Should I split the PR ?

Also this adds a new FastAPI dependency to the provider for type checking should I update requirements ? (I assume no because it has no effect at runtime but I prefer to double check)